dojo/tools/nikto/parser.py

```diff
@@ -62,86 +139,42 @@ def process_scandetail(self, scan, test, dupes):
             severity = "Info"  # Nikto doesn't assign severity, default to Info
 
             # Description
-            description = "\n \n".join((("Host: " + ip),
-                                    ("Description: " + item.find("description").text),
-                                    ("HTTP Method: " + item.attrib["method"]),
-                                    ))
-            mitigation = "N/A"
-            impact = "N/A"
-            references = "N/A"
-
-            dupe_key = hashlib.md5(description.encode("utf-8")).hexdigest()
+            description = "\n".join([
+                    f"**Host:** `{ip}`",
+                    f"**Description:** `{item.find('description').text}`",
+                    f"**HTTP Method:** `{item.attrib['method']}`",
+            ])
+
+            url = hyperlink.parse(ip)
+            endpoint = Endpoint(
+                protocol=url.scheme,
+                host=url.host,
+                port=url.port,
+                path="/".join(url.path),
+            )
+
+            dupe_key = hashlib.sha256(description.encode("utf-8")).hexdigest()
 
             if dupe_key in dupes:
                 finding = dupes[dupe_key]
                 if finding.description:
                     finding.description = finding.description + "\nHost:" + ip + "\n" + description
-                self.process_endpoints(finding, ip)
-                dupes[dupe_key] = finding
-            else:
-                dupes[dupe_key] = True
+                finding.unsaved_endpoints.append(endpoint)
+                finding.nb_occurences += 1
 
+            else:
                 finding = Finding(title=titleText,
-                                  test=test,
-                                  active=False,
-                                  verified=False,
-                                  description=description,
-                                  severity=severity,
-                                  numerical_severity=Finding.get_numerical_severity(
-                                      severity),
-                                  mitigation=mitigation,
-                                  impact=impact,
-                                  references=references,
-                                  url='N/A',
-                                  dynamic_finding=True)
+                                    test=test,
+                                    active=False,
+                                    verified=False,
+                                    description=description,
+                                    severity=severity,
+                                    numerical_severity=Finding.get_numerical_severity(
+                                        severity),
+                                    references="N/A",
```

Comment:
Why keeping this one N/A, whereas impact and and mitigation were removed?

Reply:
impact and mitigation are now optional. not sure about references.

Reply:
Optional too.

Reply:
ok let's simplify it.

Reply:
done. it's wonderful! Parsers will be more clean without every "No data here" everywhere...